tls/s2n_kem_params.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include "utils/s2n_blob.h"
+
+struct s2n_kem_params {
```

Comment:
In my opinion, both s2n_kem_params.h and s2n_kem.h shouldn't be under the tls directory. They should move to the future pq-crypto directory after PR #905 will be merged.
This will result in better consistency to other cryptographic primitives such as ECDHE and DH (found in the crypto library).
This will also provide better modularity, freeing up the tls library to deal only with TLS related aspects, making a cleaner code. 
Clearly, this modification can only take place in the future.

Meanwhile, I think that having a file for a three lines structure is a bit of a waste. I suggest merging the two files into s2n_kem.h. In addition, I suggest to rename the negotiated_kem parameter to just "kem", this fits better the separation described above.

Reply:
What about moving these to the crypto folder so it could be used in the future for any KEM that was PQ or not.

Re two files I thought about combining them but not all files that need `s2n_kem_params.h` need all of `s2n_kem.h`. Granted it all gets loaded in the end so its probably not that big of a deal.

Reply:
Will move to pq-crypto which will be the only KEMs s2n uses once https://github.com/awslabs/s2n/pull/905 or https://github.com/awslabs/s2n/pull/927 get merged in.